### PR TITLE
Modifications to fracture meshing, mainly in 2d domains

### PR DESCRIPTION
**Pull request by keileg**
1. In meshing, fractures can be marked as constraints. The 2d mesh will conform to the corresponding line, but it will not be represented as a lower-dimensional grid.
2. Cleanup of various bits of code.

### CODE REVIEW

File: src/porepy/fracs/fracture_importer.py [hunk 1]
```diff
@@ -24,6 +19,7 @@ def network_3d_from_csv(file_name, has_domain=True, tol=1e-4):
         file_name: name of the file
         has_domain: if the first line in the csv file specify the domain
         tol: (optional) tolerance for the methods
```
Comment (IvarStefansson) [inline src/porepy/fracs/fracture_importer.py]: Could we be more specific?

File: src/porepy/fracs/fractures_2d.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import csv
 
 import porepy as pp
-
+import porepy.fracs.simplex
```
Comment (IvarStefansson) [inline src/porepy/fracs/fractures_2d.py]: I notice that pp.fracs.simplex is used below

File: src/porepy/fracs/fractures_2d.py [hunk 3]
```diff
@@ -150,17 +150,35 @@ def add_network(self, fs):
...
+            constraints = np.empty(0, dtype=np.int)
 
         p = self.pts
         e = self.edges
 
         if do_snap and p is not None and p.size > 0:
             p, _ = pp.frac_utils.snap_fracture_set_2d(p, e, snap_tol=tol)
-        grid_list = pp.fracs.simplex.triangle_grid(
-            p, e[:2], self.domain, tol=tol, subdomains=constraints, **mesh_args
+        grid_list = porepy.fracs.simplex.triangle_grid(
```
Comment (IvarStefansson) [inline src/porepy/fracs/fractures_2d.py]: Is there a reason we don't import porepy as pp?

File: src/porepy/fracs/fractures_2d.py [hunk 4]
```diff
@@ -277,18 +295,21 @@ def as_graph(self, split_intersections=True):
         else:
             return G
 
-    def split_intersections(self):
+    def split_intersections(self, tol=None):
         """ Create a new FractureSet, with all fracture intersections removed
```
Comment (IvarStefansson) [inline src/porepy/fracs/fractures_2d.py]: Document tolerance

File: src/porepy/fracs/simplex.py [hunk 5]
```diff
@@ -274,7 +250,10 @@ def triangle_grid(
     return triangle_grid_from_gmsh(file_name, **kwargs)
 
 
-def triangle_grid_from_gmsh(file_name, **kwargs):
+def triangle_grid_from_gmsh(file_name, constraints=None, **kwargs):
```
Comment (IvarStefansson) [inline src/porepy/fracs/simplex.py]: Documentation

File: src/porepy/fracs/fracture_importer.py (hunk 1 above)
Comment (keileg) [reply src/porepy/fracs/fracture_importer.py]: I wish. But I can add some minor comments.

File: src/porepy/fracs/fractures_2d.py (hunk 3 above)
Comment (keileg) [reply src/porepy/fracs/fractures_2d.py]: Me struggling with imports.. I'll make a second attempt

File: src/porepy/fracs/simplex.py (hunk 5 above)
Comment (keileg) [reply src/porepy/fracs/simplex.py]: Good point. Will do.

**Pull request merged**